client/sites/settings/administration/tools/leave-site/leave-site-modal-form.tsx

```diff
@@ -0,0 +1,196 @@
+import page from '@automattic/calypso-router';
+import {
+	Button,
+	CheckboxControl,
+	Flex,
+	FlexItem,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import QuerySitePurchases from 'calypso/components/data/query-site-purchases';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { hasLoadedSitePurchasesFromServer } from 'calypso/state/purchases/selectors';
+import hasCancelableSitePurchases from 'calypso/state/selectors/has-cancelable-site-purchases';
+import { leaveSite } from 'calypso/state/sites/actions';
+import { getSite, getSiteDomain, getSiteSlug } from 'calypso/state/sites/selectors';
+import useQueryUsersMeBySiteId from './use-query-users-me-by-site-id';
+import type { AppState } from 'calypso/types';
+import './leave-site-modal-form.scss';
+
+export interface LeaveSiteModalFormProps {
+	siteId: number;
+	onSuccess?: () => void;
+	onClose?: () => void;
+}
+
+const LeaveSiteModalForm = ( { siteId, onSuccess, onClose }: LeaveSiteModalFormProps ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ isChecked, setChecked ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const wpcomUserId = useSelector( ( state: AppState ) => getCurrentUserId( state ) );
+	const siteSlug = useSelector( ( state: AppState ) => getSiteSlug( state, siteId ) );
+	const siteDomain = useSelector( ( state: AppState ) => getSiteDomain( state, siteId ) || '' );
+	const siteOwnerId = useSelector( ( state: AppState ) => getSite( state, siteId )?.site_owner );
+	const sitePurchasesLoaded = useSelector( ( state: AppState ) =>
+		hasLoadedSitePurchasesFromServer( state )
+	);
+
+	const hasActiveCancelableSubscriptions = useSelector( ( state: AppState ) =>
+		hasCancelableSitePurchases( state, siteId, wpcomUserId )
+	);
+
+	// It gets external user ID (ID of user entity from site connected via Jetpack) from provided WPCOM user ID.
+	const { data: user } = useQueryUsersMeBySiteId( siteId );
+
+	const isSiteOwner = wpcomUserId === siteOwnerId;
+
+	const handleLeaveSite = async () => {
+		if ( ! user?.id || ! isChecked ) {
+			return;
+		}
+
+		recordTracksEvent( 'calypso_leave_site_modal_form_leave_site_click' );
+		setIsSubmitting( true );
+		try {
+			const result = await dispatch( leaveSite( siteId, user?.id ) );
+			if ( result ) {
+				page.redirect( '/sites' );
+				onSuccess?.();
+				onClose?.();
+			}
+		} finally {
+			setIsSubmitting( false );
+		}
+	};
+
+	const renderBody = () => {
+		if ( hasActiveCancelableSubscriptions ) {
```

Comment:
Wait a minute, I just realized... is this check actually necessary?

- If you're an owner, you will need to transfer ownership anyway.
- If you're not an owner, then by definition you can't have purchases to the site, so you should be able to leave.

Can we double check? 🤔 🤔

Reply:
Any admin can make purchases, e.g.: Plan, Domain, Add-on, etc.

Reply:
Sorry sorry you're right; yeah I tested that and I was able to see the modal for this case! 🤦